dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -6807,7 +6958,7 @@ int getSelectPlan(gp_walk_info& gwi, SELECT_LEX& select_lex,
                     // @bug3881. set_user_var can not be treated as constant function
                     // @bug5716. Try to avoid post process function for union query.
                     if ((gwi.subQuery || select_lex.group_list.elements != 0 ||
-                            !csep->unionVec().empty() || isUnion || isPushdownHand ) &&
+                            !csep->unionVec().empty() || isUnion ) &&
```

Comment:
I missed it in the first pass. Since isPushdownHand is true, it should set the whole block in parenthesis to true.